train.py

```diff
@@ -166,9 +168,23 @@ def run_training_loop(args: argparse.Namespace) -> str:
     # load model from disk
     if args.model_weights:
         logging.info('*** loading model %s from path %s', args.model_name, args.model_weights)
-        model_weights = torch.load(args.model_weights, map_location=device)
+        model_weights = torch.load(args.model_weights, map_location=device)['model']
         logging.info('*** loaded model weights from disk, keys = %s', model_weights.keys())
-        missing_keys, unexpected_keys = experiment.model.load_state_dict(model_weights['model'], strict=False)
+
+        if args.model_weights_drop_linear:
+            # Say we want to fine-tune a previously fine-tuned model and drop its linear layers.
+            # We can do so 
+            logging.info(f'--model_weights_drop_linear set, not loading weights for linear layers')
+            linear_keys = [k for k in model_weights.keys() if k.startswith('linear')]
+
+            if not len(linear_keys):
+                raise ValueError('--model_weights_drop_linear set but no weights for linear layers found')
+
+            for key in linear_keys:
+                logging.info(f'\t- dropping weight with key `%s`', key)
+                del model_weights[key]
+
+        missing_keys, unexpected_keys = experiment.model.load_state_dict(model_weights, strict=False)
```

Comment:
Clarification: in a model where we use `--model_weights_drop_linear` , the difference between training a new model, and re-tuning a finetuned model without its linear layers is...what?  The M-matrix would be different, right?  Is that the only difference?